server/src/concepts/offer/useCases/createNewOffer.js

```diff
@@ -0,0 +1,34 @@
+import { createOffer } from "../repositories/commands.js";
+import { findCategoryByName } from "../../category/repositories/queries.js";
+import { findHeightByValue } from "../../height/repositories/queries.js";
+
+export async function createNewOffer(body) {
+  try {
+    const offer = await createOffer({
+      title: body.title,
+      description: body.description,
+      city: body.city,
+      location: body.location || {},
+      phoneNumber: body.phoneNumber,
+      photos: body.photos,
+      category: getCategoryId(body.category),
+      height: getHeightId(body.height),
+      price: body.price,
```

Comment:
what about forExchange? is false by default when is no provided, but it have to be here because it can be set when offer is creating, especially that is required by joi validation